File: ios/KhrysalisWebsite/com/lightningkite/khrysaliswebsite/vg/FaqVG.shared.swift
//Package: com.lightningkite.khrysaliswebsite.vg
//Converted using Khrysalis2

import Foundation
import Khrysalis
import RxSwift
import RxRelay



public class FaqVG: ViewGenerator {
    
    public var stack: ObservableStack<ViewGenerator>
    
    override public var title: String {
        get {
            return "Faq"
        }
    }
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = FaqXml()
        var view = xml.setup(dependency)
        xml.questions.bind(data: ConstantObservableProperty([1, 2, 3, 4]), defaultValue: 1, makeView: { (observable) in 
            var cellXml = ComponentRowQuestionXml()
            var cellView = cellXml.setup(dependency)
            cellXml.container.onClick{ () in 
                self.cellXmlContainerClick()
            }
            cellXml.question.bindString(ConstantObservableProperty("How do I do all the things?  It seems really hard and I need help"))
            return cellView
        })
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    public func cellXmlContainerClick() -> Void {
        stack.push(ArticleVG(stack: self.stack))
    }
    
    public init(stack: ObservableStack<ViewGenerator>) {
        self.stack = stack
        super.init()
    }
    convenience public init(_ stack: ObservableStack<ViewGenerator>) {
        self.init(stack: stack)
    }
}
 

File: android/src/main/java/com/lightningkite/khrysaliswebsite/vg/FaqVG.shared.kt
//
// FaqVG.swift
// Created by Khrysalis Prototype Generator
// Sections of this file can be replaces if the marker, '(overwritten on flow generation)', is left in place.
//
package com.lightningkite.khrysaliswebsite.vg

//--- Imports

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysalis.observables.*
import com.lightningkite.khrysalis.observables.binding.*
import com.lightningkite.khrysaliswebsite.R
import com.lightningkite.khrysaliswebsite.layouts.*
import com.lightningkite.khrysaliswebsite.models.Article
import com.lightningkite.khrysaliswebsite.models.Articles

//--- Name (overwritten on flow generation)
@Suppress("NAME_SHADOWING")
class FaqVG(
    //--- Dependencies (overwritten on flow generation)
    @unowned val stack: ObservableStack<ViewGenerator>
    //--- Extends (overwritten on flow generation)
) : ViewGenerator() {
    
    
    //--- Properties
    val filter = StandardObservableProperty("")
    
    //--- Title (overwritten on flow generation)
    override val title: String get() = "Faq"
    
    //--- Generate Start (overwritten on flow generation)
    override fun generate(dependency: ViewDependency): View {
        val xml = FaqXml()
        val view = xml.setup(dependency)
        
        //--- Set Up xml.filter
        xml.filter.bindString(filter)
        
        //--- Set Up xml.questions
        xml.questions.bind(
            data = filter.map { f ->
                val parts = f.split(' ')
                Articles.faq.filter { parts.all { part -> it.contains(part) } }
            },
            defaultValue = Article.empty,
            makeView = label@ { observable ->
                //--- Make Subview For xml.questions
                val cellVg = RowArticleVG(articleObservable = observable, stack = this.stack)
                val cellView = cellVg.generate(dependency)
                //--- End Make Subview For xml.questions (overwritten on flow generation)
                return@label cellView
            }
        )
        
        //--- Generate End (overwritten on flow generation)
        
        return view
    }
    
    //--- Init
    
    init {
    //--- Init End
    }
    
    //--- Actions
    
    //--- Body End
}
